### PR TITLE
Add exports field in package.json

### PR DESCRIPTION
**Pull request by QuentinDutot**
Closes https://github.com/jeffijoe/awilix/issues/368

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "awilix",
-  "version": "11.0.0",
```
Comment (jeffijoe) [inline package.json:3]: Gotta undo this since the release script will bump this

**Comment by DDuran19** [comment]
Hi team will this pull request actually fix cloudflare build issue? And just wanna ask if this will be approved (and a new NPM version be released soon? Thank you!)

**Review by jeffijoe — APPROVED** [review]

**Comment by jeffijoe** [comment]
@DDuran19 thanks for the ping, I was not notified of the push that resolved the comment! 

I've just released a new version, please test it and let me know.

**Comment by jeffijoe** [comment]
I had to revert this PR because it broke downstream consumers. It's hard to test these changes without releasing first. 😓

**Comment by DDuran19** [comment]
Hi @jeffijoe  Thanks for your immediate response! Does this also mean that the issue of deploying to cloudflare still exists and there'll be a new update coming soon to fix it?

**Comment by jeffijoe** [comment]
@DDuran19 please install version 11.0.3 and test it on CloudFlare and let me know if it works, that will help. If it works you can  continue using it until we figure out how to move forward. I'm not well-versed in all this new `exports` business so would appreciate some help here.

Basically the issue was that downstream packages such as `awilix-koa` and `awilix-express` import `awilix/lib/utils` which it didn't like after the change.

**Comment by DDuran19** [comment]
I will. I'll let you know  at the end of the day (my time) Thanks a lot~!

**Comment by jeffijoe** [comment]
@DDuran19 @QuentinDutot I've released v12.0.0 with the exports added back in, I figured out the backwards-compat issue. Please try it out!

**Comment by DDuran19** [comment]
Hi @jeffijoe  it took me a while but I have tested it and it did'nt go as planned. Apparently awilix is using multiple node-specific modules which are not available for serverless platforms like cloudflare.  for example the `const fs = require("fs")` could not be used on cloudflare because we dont have access to file systems, the requests works like a function and the context are stored in it. then after the request is resolved, everything kind of vanishes, if that makes sense.. Do you think this is something that  can be supported by awilix in the future? I really like the api, it's intuitive to use

image

image

**Comment by jeffijoe** [comment]
@DDuran19 did you `import { createContainer } from `awilix/browser`? It looks like you used the default export.

**Comment by DDuran19** [comment]
Here's what I did `import { createContainer, asClass, asFunction } from 'awilix'`

Im gonna try to import from browser :) I'll be back in a sec

**Comment by DDuran19** [comment]
@jeffijoe  I got this error: `SyntaxError: Cannot use import statement outside a module` it was still importing `import * as glob from 'fast-glob';`
image

image
image

All of those are on Dev

This is the error when building it
image

**Comment by jeffijoe** [comment]
That’s odd, the browser build should not include those

**Comment by jeffijoe** [comment]
Looks like the change in ES target broke the node-stripping part of the build, looking into this.

**Comment by jeffijoe** [comment]
@DDuran19 ok v12.0.1 should fix this! The browser build was indeed broken, the Node dependencies were not correctly shaken off. Now they should be, please try it out!

**Comment by jeffijoe** [comment]
Also, I personally wouldn't even use Awilix for serverless, but if you do, I'd recommend using `PROXY` injection mode, since the startup cost of `CLASSIC` is higher (due to having to parse the constructor/function signatures to figure out dependencies).

**Comment by DDuran19** [comment]
Thanks for all ur input! What do you recommend I use for serverless? Is the performance overhead that bad?

**Comment by jeffijoe** [comment]
Serverless is all about optimizing start times, and in `CLASSIC` mode, Awilix will need to parse the function and constructors of your registrations, which incur some overhead, although it should be extremely small. With that said, if your hosting makes it so your Node process is spawned, runs a single function, and then exits, you won't benefit from the reduced cost of subsequent resolutions. If your process stays alive and handles multiple requests, then that's a different story. I've never run serverless myself so I am not familiar with how each platform handles this.

**Comment by DDuran19** [comment]
Thanks for the clarification!

I was still having issues on `npm run dev` and `npm run build`

image

**Pull request merged**